addons/sys_logistics/fnc_setObjectCargo.sqf

```diff
@@ -55,13 +55,9 @@ _cargoI = [_cargoWMI, 2, [], [[]]] call BIS_fnc_param;
 if (isMultiplayer && {isServer}) then {_global = "Global"} else {_global = ""};
 
 // Reset Magazines state
-call (compile format["
-            {_input removeMagazine%1 _x} forEach (magazines _input);
-            {_input addMagazine [_x select 0,_x select 1]} foreach _ammo;
-        ",
-        _global
-    ]
-);
+_reset = [[{_input removeMagazine _x} forEach (magazines _input); {_input addMagazine [_x select 0,_x select 1]} foreach _ammo;],
+            [{_input removeMagazineGlobal _x} forEach (magazines _input); {_input addMagazine [_x select 0,_x select 1]} foreach _ammo;]];
+call (_reset select (_global == "Global"));
```

Comment:
The second `[` should be `{` and the `]` should become `}`. The above is wrong as far as I can tell.

Also I'd prefer something like this, for readability anyway.
```sqf
_reset = [
    {
        {_input removeMagazine _x} forEach (magazines _input);
        {_input addMagazine [_x select 0,_x select 1]} forEach _ammo;
    },
    {
        {_input removeMagazineGlobal _x} forEach (magazines _input);
        {_input addMagazine [_x select 0,_x select 1]} forEach _ammo;
    }
];
```

Though, it's probably even better to remove the entire array building and the call and replace the entire reset stuff with
```sqf
{
    if (_global == "Global") then {
        _input removeMagazineGlobal _x;
    } else {
        _input removeMagazine _x;
    };
} forEach (magazines _input);

 {_input addMagazine [_x select 0,_x select 1]} forEach _ammo;
```

Reply:
Looks a lot better. Will implement.